### ISSUE TITLE
Setting valueRange min/max to null causes exceptions

### ISSUE THREAD

**Issue by mharrington622** [IgniteUI/ignite-ui#1102]
### Description  
When setting the min or max values of the valueRange property to null, an unhandled exception will occur with certain values.

### Steps to reproduce  

1. Run [attached sample](https://github.com/IgniteUI/ignite-ui/files/1188675/page.zip).
2. In the top editor, type "5" and blur the element

### Result  
A "Cannot read property 'toString' of null" exception occurs.

### Expected result  
No exceptions should occur and the value should be valid.

### Additional Notes
The same exception occurs if you enter a negative number in the bottom editor.

The following logic in the isValid method of $.ig.igValidatorValueRule does not work well when nulls are used.

```
hasMin = typeof minValue === "number" || minValue,
hasMax = typeof maxValue === "number" || maxValue;

if ((hasMin || hasMax)) {
	if (isNumber && !options.date) {
		value = this._parseNumber(value, options);
		min = hasMin && minValue;
		min = value < min ? min.toString() : null;
		max = hasMax && maxValue;
		max = value > max ? max.toString() : null;
	}
```

**Comment by damyanpetev** [comment]
Since this can easily be avoided by setting just the case need (e.g. `valueRange: { min: 10 }`) I'd say this a is medium-low issue.

@mharrington622 Since this didn't have a version label, I'm marking it for latest. Do let me know if this came from a customer case and should be applied to previous versions.

**Comment by mharrington622** [comment]
@damyanpetev @Lipata I just realized the customer is using 15.2.  I can suggest to them that they can avoid setting the property altogether for their situation.

**Comment by Lipata** [comment]
@mharrington622 you can suggest them that, but @damyanpetev already fixed in 17.2, and as far as it is a customer request we will apply the change for both 16.2 and 17.1. Thank you!
